src/aiida_quantumespresso/workflows/xspectra/core.py

```diff
@@ -0,0 +1,910 @@
+            ),
+        )
+        spec.input(
+            'core_wfc_data',
+            valid_type=orm.SinglefileData,
+            required=False,
+            help='The core wavefunction data file extracted from the ground-state pseudo for the absorbing atom.'
+        )
+        spec.input(
+            'run_replot',
+            valid_type=orm.Bool,
+            serializer=to_aiida_type,
+            default=lambda: orm.Bool(False),
+        )
+        spec.input(
+            'upf2plotcore_code',
+            valid_type=orm.Code,
+            serializer=to_aiida_type,
+            required=False,
+            help='The code node required for upf2plotcore.sh configured for ``AiiDA-Shell``. '
+            'Must be provided if `core_wfc_data` is not provided.'
+        )
+        spec.input(
+            'clean_workdir',
+            valid_type=orm.Bool,
+            serializer=to_aiida_type,
+            default=lambda: orm.Bool(False),
+            help=('If `True`, work directories of all called calculation will be cleaned at the end of execution.'),
+        )
+        spec.input(
+            'dry_run',
+            valid_type=orm.Bool,
+            serializer=to_aiida_type,
+            required=False,
+            help='Terminate workchain steps before submitting calculations (test purposes only).'
+        )
+        spec.outline(
+            cls.setup,
+            cls.run_scf,
+            cls.inspect_scf,
+            if_(cls.should_run_upf2plotcore)(
+                cls.run_upf2plotcore,
+                cls.inspect_upf2plotcore,
+            ),
+            cls.run_all_xspectra_prod,
+            cls.inspect_all_xspectra_prod,
+            if_(cls.should_run_replot)(
+                cls.run_all_xspectra_plot,
+                cls.inspect_all_xspectra_plot,
+            ),
+            cls.results,
+        )
+
+        spec.exit_code(401, 'ERROR_SUB_PROCESS_FAILED_SCF', message='The SCF sub process failed')
+        spec.exit_code(402, 'ERROR_SUB_PROCESS_FAILED_XSPECTRA', message='One or more XSpectra sub processes failed')
+        spec.exit_code(403, 'ERROR_NO_GIPAW_INFO_FOUND', message='The pseudo for the absorbing element contains no'
+                       ' GIPAW information.')
+        spec.output('output_parameters_scf', valid_type=orm.Dict, help='The output parameters of the SCF'
+                    ' `PwBaseWorkChain`.')
+        spec.output_namespace(
+            'output_parameters_xspectra',
+            valid_type=orm.Dict,
+            help='The output dictionaries of each `XspectraBaseWorkChain` performed',
+            dynamic=True
+        )
+        spec.output(
+            'output_spectra',
+            valid_type=orm.XyData,
+            help='An XyData node containing all the final spectra produced by the WorkChain.'
+        )
+        spec.output('powder_spectrum', valid_type=orm.XyData, required=False, help='The simulated powder spectrum')
```

Comment:
Let's just get rid of the `output_` prefix for all the outputs. We still have this in some other workchains for historical reasons that we cannot change because of backwards compatibility, but it is redundant. If you do `node.outputs.xspectra` it is clear that is an output and you don't need `node.outputs.output_xspectra` to tell you.